### ISSUE TITLE
Getting an AssertionError when attempting to make variable fonts

### ISSUE THREAD

**Issue by EwonRael** [googlei18n/fontmake#396]
Hey,

I'm having trouble making a variable font right now. I have gotten similar errors in the past but was still able to generate a variable font. I try to generate the font using the `fontmake -o variable -m fontmakefile` command and get the following error:

> Traceback (most recent call last):
>   File "/env/bin/fontmake", line 11, in <module>
>     load_entry_point('fontmake', 'console_scripts', 'fontmake')()
>   File "/fontmake/Lib/fontmake/__main__.py", line 191, in main
>     project.run_from_designspace(designspace_path, **args)
>   File "/fontmake/Lib/fontmake/font_project.py", line 437, in run_from_designspace
>     interpolate_layout_from=interpolate_layout_from, **kwargs)
>   File "/example/fontmake/Lib/fontmake/font_project.py", line 496, in run_from_ufos
>     self.build_variable_font(designspace_path)
>   File "/example/fontmake/Lib/fontmake/font_project.py", line 213, in build_variable_font
>     assert all(p.startswith(ufodir) for p in ufo_paths)
> AssertionError

I assumed it had something to do with the install so I installed fontmake in a virtual environments and made sure to run `pip install -r dev_requirements.txt` but am still getting the same error. This seems to be related to [the problem encountered by @marekjez86](https://github.com/googlei18n/fontmake/issues/386), but reading through that it was unclear to me what the solution is.

Thanks!

**Comment by miguelsousa** [comment]
This is a longstanding problem. **fontmake** requires the source files to be organized in a certain way – I'm not even sure what the right way is supposed to be. That particular `assert` is to guarantee that all source UFO files are stored in the same directory.

I ran into this same limitation when I tried to make variable TTFs of [Source Sans](https://github.com/adobe-fonts/source-sans-pro) and [Source Serif](https://github.com/adobe-fonts/source-serif-pro). My solution was to make some [modifications to `font_project.py`](https://github.com/adobe-type-tools/fontmake/commit/334d215a8e4dcfa4750ea3d6de89be02a1a8e496).

I'd prefer to NOT use a fork, but I haven't found a chance to submit a PR here.

/cc @anthrotype @behdad

**Comment by EwonRael** [comment]
Right-o! Thanks for the quick reply and cohesive explanation.

I redid the organization of my source files so that they're all in the same place, reran fontmake, and ran into the following error

> IOError: [Errno 2] No such file or directory: 'PathToSourceFile/./sources/font-0.ttf'

I looked it over and it appears that previous version of fontmake put the generated interpolatable font files in the source directory, but the newer version puts them in a "master_ttf_interpolatable" folder. However, fontmake still *looks* for the source files in the source directory. 

A quick fix is to move all the outputted .ttf files into the source directory and rerun fontmake, which I did and it ran without issue! For my sake this is a usable work-flow, but it is an obvious issue.
